core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -1037,71 +1037,75 @@ public void testCatalogTokenRefresh() throws Exception {
     catalog.initialize(
         "prod", ImmutableMap.of(CatalogProperties.URI, "ignored", "credential", "catalog:secret"));
 
-    Thread.sleep(3_000); // sleep until after 2 refresh calls
```

Comment:
there are no code differences in the test code itself, we're just wrapping it in a 
```
Awaitility.await()
        .atMost(5, TimeUnit.SECONDS)
        .untilAsserted(
            () -> { ... })
```